source/common/upstream/eds.cc

```diff
@@ -48,17 +48,19 @@ void EdsClusterImpl::startPreInit() { subscription_->start({cluster_name_}); }
 
 void EdsClusterImpl::BatchUpdateHelper::batchUpdate(PrioritySet::HostUpdateCb& host_update_cb) {
   absl::node_hash_map<std::string, HostSharedPtr> updated_hosts;
```

Comment:
As long as you are in here can you make this flat_hash_map? I don't see why we would need node stability.

Reply:
updated_hosts is used in several places, including as a parameter to BaseDynamicClusterImpl::updateDynamicHostList and it's moved to all_hosts_. This resulted in needing to change a few things.

In the end I decided that it might be easier to change the definition of Upstream::HostMap. I think doing that is safe, but can easily revert it if you'd rather I didn't.